### PR TITLE
Revise squared difference

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - clean FusedOp legacy from squared_difference operation
 - add type_prop tests
 - add backend tests

### Tickets:
 - 44625

### CODE REVIEW

File: ngraph/core/include/ngraph/op/squared_difference.hpp [hunk 1]
```diff
@@ -4,9 +4,7 @@
 
 #pragma once
 
-#include "ngraph/node.hpp"
-#include "ngraph/op/op.hpp"
-#include "ngraph/op/util/fused_op.hpp"
+#include "ngraph/op/util/binary_elementwise_arithmetic.hpp"
 
 NGRAPH_SUPPRESS_DEPRECATED_START
```
Comment (jdanieck) [inline ngraph/core/include/ngraph/op/squared_difference.hpp:9]: Shouldn't we remove this deprecated macros?

File: ngraph/test/type_prop/square_difference.cpp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Copyright (C) 2018-2021 Intel Corporation
```
Comment (jdanieck) [inline ngraph/test/type_prop/square_difference.cpp:1]: // Copyright (C) 2021 Intel Corporation

File: ngraph/test/backend/squared_difference.in.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2018-2021 Intel Corporation
```
Comment (jdanieck) [inline ngraph/test/backend/squared_difference.in.cpp:1]: // Copyright (C) 2021 Intel Corporation

**Review by jdanieck — APPROVED** [review]

File: ngraph/test/type_prop/square_difference.cpp (hunk 2 above)
Comment (bszmelcz) [reply ngraph/test/type_prop/square_difference.cpp]: I found out that I missed spell the file name, and the type_prop/square_difference.cpp file was already created containing single test. So I changed it  to match the new formula, but now I dont know if I should leave the old date 2018-2021 or still change it to 2021

Comment (jdanieck) [reply ngraph/test/type_prop/square_difference.cpp]: If file existed then the old date.

File: ngraph/core/include/ngraph/op/squared_difference.hpp [hunk 4]
```diff
@@ -19,40 +15,29 @@ namespace ngraph
             /// \brief Calculates an element-wise squared difference between two tensors
             ///
             /// y[i] = (x1[i] - x2[i])^2
-            class NGRAPH_API SquaredDifference : public ngraph::op::util::FusedOp
+            class NGRAPH_API SquaredDifference : public util::BinaryElementwiseArithmetic
             {
             public:
                 static constexpr NodeTypeInfo type_info{"SquaredDifference", 0};
                 const NodeTypeInfo& get_type_info() const override { return type_info; }
```
Comment (GlebKazantaev) [inline ngraph/core/include/ngraph/op/squared_difference.hpp:22]: please update with new ngraph RTTI.

**Comment by bszmelcz** [comment]
@ilyachur @jdanieck I am writing this comment to open the dicussion if some changes should be applied to the legacy test op::is_binary_elementwise_arithmetic() since I guess that currently it is not working proprerly since I assume that squared_difference operator should be of this type. Do you have any ideas?

**Review by ilyachur — APPROVED** [review]

**Pull request merged**